Unable to compile with futures-test as dev-dependency

Adding `futures-test-preview = "0.3.0-alpha.11"` to dev-dependencies causes the following compilation error for me:

```log
error[E0425]: cannot find function `block_on` in module `futures_executor`
  --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-test-preview-0.3.0-alpha.11/src/future/mod.rs:79:44
   |
79 |         thread::spawn(|| futures_executor::block_on(self));
   |                                            ^^^^^^^^ not found in `futures_executor`

error: aborting due to previous error
```

I think this is the case because by default it doesn't enable the `std` features in futures-executor, see https://github.com/rust-lang-nursery/futures-rs/blob/71c59479b538380567907b115ec71799eb03b88d/futures-test/Cargo.toml#L20

Which means `block_on` isn't available, see https://github.com/rust-lang-nursery/futures-rs/blob/71c59479b538380567907b115ec71799eb03b88d/futures-executor/src/lib.rs#L14-L15


I assume the fix is as simple as enabling the `std` feature on future-executor by default, but I don't what other things that would cause.

--- a/futures-test/Cargo.toml
+++ b/futures-test/Cargo.toml
@@ -22,3 +22,7 @@ pin-utils = { version = "0.1.0-alpha.4", default-features = false }
 
 [dev-dependencies]
 futures-preview = { version = "0.3.0-alpha.2", path = "../futures", default-features = false, features = ["std"] }
+
+[features]
+default = ["std"]
+std = ["futures-core-preview/std", "futures-util-preview/std", "futures-executor-preview/std"]

--- a/futures-test/src/lib.rs
+++ b/futures-test/src/lib.rs
@@ -11,16 +11,24 @@
     html_root_url = "https://rust-lang-nursery.github.io/futures-doc/0.3.0-alpha.5/futures_test"
 )]
 
+#[cfg(not(feature = "std"))]
+compile_error!("`futures-test` must have the `std` feature activated, this is a default-active feature");
+
 #[doc(hidden)]
+#[cfg(feature = "std")]
 pub use std as std_reexport;
 
 #[doc(hidden)]
+#[cfg(feature = "std")]
 pub extern crate futures_core as futures_core_reexport;
 
 #[macro_use]
 #[doc(hidden)]
+#[cfg(feature = "std")]
 pub mod assert;
 
+#[cfg(feature = "std")]
 pub mod task;
 
+#[cfg(feature = "std")]
 pub mod future;

